### ISSUE TITLE
darktable intermittently - NAS connection

### ISSUE THREAD

**Issue by Hal-RawPhotographyTutorials** [darktable-org/darktable#13569]
Hi,

darktable 4.2.0 
Manjaro Plasma kernel 6.1.9-1
RAM: 32G
NVIDIA GeForce GTX970 4G
OpenCL installed and active

3 collections - 2 on the NAS (connected via NFS) and one on the local disk.
First NAS collection (NASa) has 56 folders and 13300 photos. This one causes no problems
Second NAS collection (NASb) has 265 folders and 356000 photos. This is the troublesome collection. will explain the problem below
Local collection has a small variable number of folders (less than 10) and photos. I mainly use it for staging. I import here first from cards, go through the photo, rate them, delete them and process/export some. I move them to one of the NAS collections once I am done with them for the time being.

dartable works flawlessly if I only have NASa attached. However, once I import NASb, it starts hanging randomly for variable amont of time ranging from 1-2 secs to 20-30 (est. I did not actually time it). This happens even when I am working on the local collection.

Both of the NAS folders are mapped using the same configuration/mount options on both the NAS and the desktop. The NAS and desktop are both on a wired LAN connected through a single router.
I tried deleting all sidecar files from NASb and reimporting the collection, the problem persists. The problem disappears immediately if I remove NASb collection.
I noticed the following in the terminal when darktable hangs:
```
637,643112 [sql] /usr/src/debug/darktable/darktable-4.2.0/src/common/film.c:525, function dt_film_set_folder_status(): prepare “DELETE FROM memory.film_folder”
637,643979 [sql] /usr/src/debug/darktable/darktable-4.2.0/src/common/film.c:531, function dt_film_set_folder_status(): prepare “SELECT id, folder FROM main.film_rolls”
637,644032 [sql] /usr/src/debug/darktable/darktable-4.2.0/src/common/film.c:536, function dt_film_set_folder_status(): prepare “INSERT INTO memory.film_folder (id, status) VALUES (?1, ?2)”
```

and the following once it starts responding again:
`769,254071 [sql] /usr/src/debug/darktable/darktable-4.2.0/src/libs/collect.c:1395, function tree_view(): prepare “SELECT folder, film_rolls_id, COUNT(*) AS count, status FROM main.images AS mi JOIN (SELECT fr.id AS film_rolls_id, folder, status FROM main.film_rolls AS fr JOIN memory.film_folder AS ff ON fr.id = ff.id) ON film_id = film_rolls_id WHERE (1=1) GROUP BY folder, film_rolls_id”`

Though I am not sure if that is the case every time or not.
The part that puzzles me the most is why would the problem appear even when I am not working on the troublesome NAS share.

I tried a fresh install of darktable after deleting cache and config - no improvement

**Comment by TurboGit** [comment]
> dartable works flawlessly if I only have NASa attached. However, once I import NASb, it starts hanging randomly for variable amont of time ranging from 1-2 secs to 20-30 (est. I did not actually time it). This happens even when I am working on the local collection.

You mean that just mounting the NAS on your box and changing nothing else (still working on the local collection) makes darktable slow?

A question, do you confirm that all your library.db/data.db are on local drive?

**Comment by Hal-RawPhotographyTutorials** [comment]
Not just mounting the NAS on the desktop but adding the collection from the NAS in darktable.

yes both are in ~/.config/darktable on the local HDD

**Comment by ralfbrown** [comment]
Looking at the code for `dt_film_set_folder_status`, what it is doing is scanning every folder listed in the database to check whether it is still present (accessible).  This would be used to control whether the folder name is displayed with a strike-through in the collections module.

On a NAS, this could indeed be quite slow if the directory tree is not fully cached in RAM by the operating system.

Of the two places this function is called, one is a callback for a changed mount (function `_mount_changed` in src/libs/collect.c) which rebuilds the entire tree/list view after calling `dt_film_set_folder_status`.  The rebuild function is what outputs your "after" message, so this is probably the call which is taking so long.

**Comment by Hal-RawPhotographyTutorials** [comment]
Is that the expected behaviour? darktable keeps on scanning every folder every few minutes even when I am in darkroom working on a local file?

**Comment by ralfbrown** [comment]
No, that callback function should only be invoked when the operating system signals a change.  So is there some reason for the OS to do so?
The only thing I can think of is that there is some auto-mount process running that constantly remounts the drive.

**Comment by wpferguson** [comment]
I use a Synology NAS to house part of my collection.  The disks spin down after an amount of time when it's not being accessed.  So when I have it mounted and I access a folder that's on the NAS I have to wait until the NAS spins up.

**Comment by Hal-RawPhotographyTutorials** [comment]
> No, that callback function should only be invoked when the operating system signals a change. So is there some reason for the OS to do so? The only thing I can think of is that there is some auto-mount process running that constantly remounts the drive.

Great that pointed me in the correct direction. I increased the timeout on the NFS mount and that seems to have fixed it.
I appreciate the help!
